### DISCUSSION TITLE
I use next  "js" after learning typescript and I noticed vs code didn't show any error when I missed to import

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#49136] [Help] [answered]
### Summary

I am not sure what is happening . I want to see error in development when using js  , example when I forget to import my custom state , it should tell me with error . But I only know after calling that state , if not calling it,  I will never know it have error . So is this normal in JS ?
Here is example of no showing error when import is missing 

![Screenshot (134)](https://user-images.githubusercontent.com/71011043/235892310-f97b8873-fcb5-4f18-9d85-2602f4e69d0b.png)
![Screenshot (135)](https://user-images.githubusercontent.com/71011043/235892804-e7792832-97cd-4e7f-a5ca-985fdca7b088.png)

**Comment by icyJoseph** [comment, marked as answer]
JavaScript is an interpreted language. Actual runtime errors do not show up until the badly defined line of code is run, so to speak.

```jsx
export default function App() {
  return (
    <div>
      <button
        onClick={() => {
          iDoNotExist();
        }}
      >
        I crash the app
      </button>
    </div>
  );
}
```

Typically for this case, you kind of want to have ESLINT's no-undef, activated: https://eslint.org/docs/latest/rules/no-undef

In an app with no ESLINT, and JS/JSX files, I think VSCode, won't say anything at all. 

With ESLINT enabled, for instance using this config:

```jsx
{
  "extends": "next/core-web-vitals",
  "rules": {
    "no-undef": "error"
  }
}
```

Then I'd see, on VSCode, an error, saying that `iDoNotExist` is not defined.

You can also add this magic comment `// @ts-check`:

```jsx
// @ts-check

export default function Home() {
  return (
    <>
      <div>
        <button
          onClick={() => {
            iDoNotExist();
          }}
        >
          I crash the app
        </button>
      </div>
    </>
  );
}
```

And it'll start to complain as well. I thin you might need to have a `jsconfig.json` file with `jsx`, set to "presever" IIRC.

![Screenshot 2023-05-03 at 13 08 38](https://user-images.githubusercontent.com/21013447/235899793-f3870dba-6fd9-4cb9-a84e-94eb2d05a6e8.png)


Screenshot shows both approaches, ESLINT no-undef, and ts-check, compared to no checks at all:

![Screenshot 2023-05-03 at 13 10 40](https://user-images.githubusercontent.com/21013447/235900288-334a8a33-7d20-4d39-bcf6-51014a731cdc.png)

**Reply by thanhtutzaw** [reply]
### It works now after adding rule undef in eslint.json
### but Can I also see error in terminal ? 
 
 
![Screenshot (139)](https://user-images.githubusercontent.com/71011043/236092014-afbf9ef6-1f0b-4c0b-8634-9e9ec8f704b0.png)

**Reply by icyJoseph** [reply]
The thing to understand here is that these processes, Type Checking or ESLINT, have to be run. In VSCODE there's a dedicated TypeScript, and ESLINT servers, which parse your files.

Now out from the terminal (even if you use the built-in VSCode terminal), we are in a different process. AFAIK, running `next dev` doesn't really start the linter, but I think the `next build` command does.

Typically people do a `next lint` run, to get an overview of the entire project: https://nextjs.org/docs/basic-features/eslint

**Comment by thanhtutzaw** [comment]
Do I still need to set rule "no undef" in eslint.json

![Screenshot (138)](https://user-images.githubusercontent.com/71011043/236091391-0d39f3b0-4341-4528-8b7b-8e8811a49927.png)

**Reply by icyJoseph** [reply]
If you want to get the linter to warn you about it, yes, at least with the versions I tested. ESLINT can extend base configurations, and if those change, it might require you to do some changes of your own. You can also override those base configurations. A popular base configuration is AirBnB's, https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb
